Adding meeting notes for Editors meeting 44 (#261)
cardano-foundation/CIPs: BiweeklyMeetings/2022-05-10.md

Okay.

**Matthias**-:
Yes, indeed. And as is saying in the chat, there is a lot of controversy over whether you should be able to add royalties to a project which did not need them within the original policy. And this is indeed a topic that's been discussed in the past here. So I think this is part of what the thread will be about. People disagreeing about whether or not a CIP makes sense in itself, but it's also two separate concerns in a way. That a CIP in the end is just a proposal always. And whether or not these proposals end up being adopted by marketplaces and projects, that's a different topic. So if the proposal makes sense from a technical standpoint and it's something that could indeed be implemented and not lead to security issues or stuff like that, that makes for valid proposal. And you may still disagree with the proposal and not implement it, that doesn't discard the proposal from being proposed. But if there is indeed no proposals, then we can just close it. Maybe for just a discussions, like have an idea. What do you think? Bad idea. Okay. Close it.

**M. Àngels**-:
Do you make any comment on what Matthias just commented or shall we move forward?

**Matthias**-:

> Only to agree. 

Reviewer: my guess is that Sebastien would have said this (?)

Reviewer 2: Casually agreeing with myself :joy:

**Matthias**-:
In CIP 4, what key is used here to compute the checksum? Is it directly coming from...

**Sebastien**-:
The public key.

**Matthias**-:
The root public key as well?

**Sebastien**-:
Not the root public keys, it's the account public key.

**Matthias**-:
Okay. The account public key. Okay. So you have one checksum and per account, not per wallet in the end.

**Sebastien**-:
Yep. Because the main reason that was created was to make sure that users are restoring the right wallet. [inaudible 00:43:15] the account they were restoring is correct.

**Matthias**-:
Yeah. But CIP 30 operates at the level of wallets.

**Sebastien**-:
No, it's still account based.

**Matthias**-:
Still account based?

**Sebastien**-:
You're getting addresses for an account. You sign with an account.

**Matthias**-:
And okay. Then it's up to the wallet to decide which account I guess, because there is nothing in API that's that lets you select the account.

**Sebastien**-:
Yeah. It's when you enable CIP 30 the wallet will show you a dropdown asking you which account you want.

**Matthias**-:
Okay. So it could still work for that then.

**Sebastien**-:
Yeah.

**Matthias**-:

> Okay. The idea is clear, but there is the question of making it a proper extension to CIP 30. At the same time, last time we said, we might want to sort of freeze the CIP 30 as it is now and have new extensions come as new CIPs just for the sake of having one core, which is fixed, but this thing feels also quite core in the end. So it could be just added before we freeze everything. Yeah. Okay. So from me, at least I would be quite in favor of extending the CIP 30 with a get wallet ID or get wallet checksum functions that would implement CIP 4, and have that as a the wallet ID, if it's ever needed by a DApp for any form of identification. Hmm. It's interesting 204, but we cannot actually do that with add to 5519. We can do it to convert to 5519. So we need to convert or translate the key, which is not unfeasible, but it was not quite recommended back in the days.

Reviewer: ```suggestion
Okay. The idea is clear, but there is the question of making it a proper extension to CIP 30. At the same time, last time we said, we might want to sort of freeze the CIP 30 as it is now and have new extensions come as new CIPs just for the sake of having one core, which is fixed, but this thing feels also quite core in the end. So it could be just added before we freeze everything. Yeah. Okay. So from me, at least I would be quite in favor of extending the CIP 30 with a `getWalletId` or `getWalletChecksum` functions that would implement CIP 4, and have that as a the wallet ID, if it's ever needed by a DApp for any form of identification. Hmm. It's interesting 204, but we cannot actually do that with add to 5519. We can do it to convert to 5519. So we need to convert or translate the key, which is not unfeasible, but it was not quite recommended back in the days.
```
... not recommending we must always do this, because it could add a lot of manual work to preparing the minutes... only that it's an additional thing we might check for in the transcriptions 🤔

**Matthias**-:
Yeah. I'm betting markdown files like that, as it, sorry... JSON files like that as a markdown table could be nice, but yeah, definitely if we want to do generation, that's easier in this direction from JSON to markdown than from markdown to JSON. Okay.

**Sebastien**-:
Yeah. But mostly I would want to check, we don't have any CI resources to generate the markdown from JSON. If we remove the mark down and replace with JSON OEM and generate the markdown automatically.

**Matthias**-:
Okay. I mean, fine like that, but just, or just comment on that. Okay. So yeah, I think we can just proceed with these two. I will do it after the call. Okay. What do we have next?

 #### PR181
[PR181 CIP30: getUtxos null instead of undefined](https://github.com/cardano-foundation/CIPs/pull/181)

**Sebastien**-:
Yeah, this is a bug in the current CIP 30 spec. Basically we right now specify that the function returns undefined in some cases, but undefined is not a valid JSON value. You can only use null. So basically changes.

**Matthias**-:

> Oh. And you can only serialize JSON, right? Through the web bottler API.

Reviewer 2: ```suggestion
Oh. And you can only serialize to JSON, right? Through the web browser API.
```

**Sebastien**-:
Yes.

**Matthias**-:
Doh.

**Sebastien**-:
So if you pass undefined, it gets silently converted to null. So just to avoid people making bugs just to change this back, please.

**Matthias**-:
Okay. I can't find it though. It's what number again? It's sorry. It should be the agenda. Yes. But why can I not access it? Ah.

**M. Àngels**-:
You can not access the agenda?

**Matthias**-:

> No, it was me. I was putting an extra S in the URL. Yes. Well, I wish we had proper send types, but I guess we won't.

Reviewer 2: ```suggestion
No, it was me. I was putting an extra S in the URL. Yes. Well, I wish we had proper sum-types, but I guess we won't.
```

**Sebastien**-:
Yeah. I think this API is kind of ugly anyway, but you know, we can fix it in a separate PR, at this point we can just...

**Matthias**-:
Yeah. Yeah. And there has been discussion recently as well with this wallet connect API, right. This recent and NFT launch of Instagram which uses it slightly...

**Sebastien**-:
Yeah. But that will work very differently from CIP 30.

**Matthias**-:

> Yeah. It's a bit of us spirit though. Oh, except you have this extra bridge in the middle and the way things work on Cardano are a bit different, but maybe there is also room for alignment a bit. I'm not sure if you would simplify anything for external people to integrate. I think CIP 30 is mostly fine as it is, except maybe the versioning part that we mentioned last time, which I'm still willing to make a pull request for.

Reviewer 2: ```suggestion
Yeah. It's a bit of the same spirit though. Oh, except you have this extra bridge in the middle and the way things work on Cardano are a bit different, but maybe there is also room for alignment a bit. I'm not sure if you would simplify anything for external people to integrate. I think CIP 30 is mostly fine as it is, except maybe the versioning part that we mentioned last time, which I'm still willing to make a pull request for.
```

**Sebastien**-:
Yeah. But in the meantime, I think we can merge this PR because it's just a small fix.

#### PR182
[PR182 CIP-1852: fix broken CIP links](https://github.com/cardano-foundation/CIPs/pull/182)

**Matthias**-:

> Yeah. It's fixing a inconsistency. So next one, PR 182. No, what is it? Yeah. 182. Which is fixing some login links yeah. Was approved a while back and we could just go for it. So the issue is that some of the CIPs reference other CIPs sometimes using relative links, which works on GitHub, but don't necessarily work on, on the website. So it's slightly better to use absolute URLs to make sure that, you know, things works on all platforms. We discussed that briefly on the hue and it's only about one of the CIPs anyway, but that's something we should be a bit careful with when merging new CIPs as well. So we've approved it. So unless someone has anything to say, also I'll just go and merge it. Okay. All good then.

Reviewer 2: ```suggestion
Yeah. It's fixing a inconsistency. So next one, PR 182. No, what is it? Yeah. 182. Which is fixing some broken links yeah. Was approved a while back and we could just go for it. So the issue is that some of the CIPs reference other CIPs sometimes using relative links, which works on GitHub, but don't necessarily work on the website. So it's slightly better to use absolute URLs to make sure that, you know, things works on all platforms. We discussed that briefly on the issue and it's only about one of the CIPs anyway, but that's something we should be a bit careful with when merging new CIPs as well. So we've approved it. So unless someone has anything to say, also I'll just go ahead and merge it. Okay. All good then.
```

**M. Àngels**-:
Now I have added to the agenda, some issues. Do you want to review this or do you want to leave them for after the discussion point, which are the CIP 50 and the one you brought at the beginning?

**Matthias**-:
Yeah. Let's first talk about CIP 50 and the audit guidelines first so that we have time. And then if time is left, we can go for issues, which are a bit more minor points.

**M. Àngels**-:
Okay.

**Matthias**-:
Do we have Kevin today or not? No...

**M. Àngels**-:
We have Michael but not Kevin.

### Last Check 

### Review 

### Discussions 

#### PR242
 [PR242 CIP-0050? | Shelleys Voltaire decentralization update](https://github.com/cardano-foundation/CIPs/pull/242)

**Matthias**-:

> Yeah. So maybe Michael knows, actually. So there was this discussion a few weeks ago to start a study group on CIP 50 with people from IOG and this was something Kevin picked up and was sorting. I'm not sure if it has happened yet or no. It hasn't happened yet Michael, okay. So that's something we'd need to follow up with. I know Kevin did reach out to a few people and it was acting behind the scene to get it done, but yeah, I will just reach out to him. Yes, definitely. Vasil has brought some attention already. But yeah, there are people I think we can, from which we can pick their brain to just stop thinking about that who are not directly involved in Vasil. Okay. I will just reach out to Kevin and ask him if he has any update on that regarding this study group. Maybe you can bring Michael on screen, if just for a short of date, maybe you have had also some discussions on the site already or got any feedback that you want to share?

Reviewer 2: ```suggestion
Yeah. So maybe Michael knows, actually. So there was this discussion a few weeks ago to start a study group on CIP 50 with people from IOG and this was something Kevin picked up and was sorting. I'm not sure if it has happened yet or no. It hasn't happened yet Michael, okay. So that's something we'd need to follow up with. I know Kevin did reach out to a few people and it was acting behind the scene to get it done, but yeah, I will just reach out to him. Yes, definitely. Vasil has brought some attention already. But yeah, there are people I think we can, from whom we can pick their brain to just start thinking about that who are not directly involved in Vasil. Okay. I will just reach out to Kevin and ask him if he has any update on that regarding this study group. Maybe you can bring Michael on screen, if just for a short time, maybe you have had also some discussions on the side already or got any feedback that you want to share?
```

**Michael**-:
All right, here we go. There's always many discussions on the side about this particular CIP, the very important CIP in many ways. So this will be continued to be [inaudible 00:16:12] feedback of individuals including yourself.

**Matthias**-:

> Yeah. Okay. So I guess this is also one of the CIPs where it would be worth maybe giving some summary of discussions every week. So I haven't gone through the whole discussion thread I admit yet, but I will try to yeah... Post summaries a bit like we did for the... Oh, what was it? The NFT mid standards. And CIP 30 actually, back in the days. So we get some yeah, regular point of synchronization where we can move the discussions a bit forwards.

Reviewer 2: ```suggestion
Yeah. Okay. So I guess this is also one of the CIPs where it would be worth maybe giving some summary of discussions every week. So I haven't gone through the whole discussion thread I admit yet, but I will try to yeah... Post summaries a bit like we did for the... Oh, what was it? The NFT standards. And CIP 30 actually, back in the days. So we get some yeah, regular point of synchronization where we can move the discussions a bit forwards.
```

**M. Àngels**-:
Okay.

**Matthias**-:
Yes, indeed. And as is saying in the chat, there is a lot of controversy over whether you should be able to add royalties to a project which did not need them within the original policy. And this is indeed a topic that's been discussed in the past here. So I think this is part of what the thread will be about. People disagreeing about whether or not a CIP makes sense in itself, but it's also two separate concerns in a way. That a CIP in the end is just a proposal always. And whether or not these proposals end up being adopted by marketplaces and projects, that's a different topic. So if the proposal makes sense from a technical standpoint and it's something that could indeed be implemented and not lead to security issues or stuff like that, that makes for valid proposal. And you may still disagree with the proposal and not implement it, that doesn't discard the proposal from being proposed. But if there is indeed no proposals, then we can just close it. Maybe for just a discussions, like have an idea. What do you think? Bad idea. Okay. Close it.

**M. Àngels**-:
Do you make any comment on what Matthias just commented or shall we move forward?

> **Matthias**-:

Reviewer 2: ```suggestion
**Sebastien**-:
```
